cpp/src/io/orc/orc_gpu.h

```diff
@@ -85,6 +85,14 @@ struct DictionaryEntry {
  */
 constexpr int orc_decimal2float64_scale = 0x80;
 
+/**
+ * Struct to store parent column mask and null count
+ */
+struct ParentColumnData {
```

Comment:
Nothing in this class is specific to "parent" columns. IMO this could be called something like `column_validity_data` or `column_validity_info`.